Add documentation for the sqlite backend options (#14239)
gravitational/teleport: docs/pages/setup/reference/backends.mdx

postgres=# CREATE DATABASE teleport;
postgres=# GRANT ALL PRIVILEGES ON DATABASE teleport TO dbuser;
```

## SQLite

The SQLite backend is used when no `type` is specified in the storage section in
the Teleport configuration file, or when `type` is set to `sqlite` or `dir`.
It's not designed for high throughput and it's not capable of serving the needs
of Teleport's High Availability configurations; as such, it's only recommended

> for deployments of moderate size.

Author: I started adding something along the lines of

```
The load on the backend depends on both how many resources are connected to the
cluster and how active said cluster is, and how well SQLite can cope with the load depends heavily on the IO resources available to the machine running the Auth Service, so we can't give any numbers here, but a sign that [...]
```

but that feels even worse than saying nothing, somehow.

A sign that the cluster has outgrown the capabilities of the SQLite backend in a
local directory is the presence of warning messages in the auth server's logs

> that say `SLOW TRANSACTION`. As a stopgap measure until it's possible to migrate

Reviewer: ```suggestion
that say `SLOW TRANSACTION`. 

As a stopgap measure until it's possible to migrate
```
The paragraph is a little long

> the cluster to use a HA-capable backend, the SQLite backend can be configured to

Reviewer: ```suggestion
the cluster to use a HA-capable backend, you can configure the SQLite backend to
```
to use the active voice

reduce the amount of disk synchronization, in exchange for less resilience
against system crashes or power loss. For an explanation on what the options
mean, see [the official SQLite
docs](https://www.sqlite.org/pragma.html#pragma_synchronous). No matter the
configuration, we recommend you take regular backups of your cluster state.

> ```yaml

Reviewer: I think we need to connect these two code snippets to the preceding paragraph more explicitly.

postgres=# CREATE DATABASE teleport;
postgres=# GRANT ALL PRIVILEGES ON DATABASE teleport TO dbuser;
```

## SQLite

The Auth Service uses the SQLite backend when no `type` is specified in the
storage section in the Teleport configuration file, or when `type` is set to
`sqlite` or `dir`. It's not designed for high throughput and it's not capable of
serving the needs of Teleport's High Availability configurations; as such, it's

> only recommended for deployments of moderate size.

Reviewer: To make the scaling recommendations more specific and actionable, what would you think of:

- Removing "; as such, it's
only recommended for deployments of moderate size"
- In the next paragraph, add a sentence similar to, "If you are planning to use SQLite as your backend, scale your cluster slowly and monitor the number of `SLOW TRANSACTION` logs".

> A sign that the cluster has outgrown the capabilities of the SQLite backend in a
> local directory is the presence of warning messages in the Auth Service's logs

Reviewer 2: ```suggestion
A sign that the cluster has outgrown the capabilities of the SQLite backend is the presence of warning messages in the Auth Service's logs
```
